bqplot/nbextension/Figure.js

```diff
@@ -404,13 +405,20 @@ define(["nbextensions/widgets/widgets/js/widget", "./components/d3/d3", "base/js
             this.svg.attr("width", null);
             this.svg.attr("viewBox", "0 0 " + this.model.get("min_width") +
                                         " " + this.model.get("min_height"));
-            setTimeout(_.bind(this.update_layout2, this), 0);
+            var that = this;
+            return new Promise(function(resolve, reject) {
+                setTimeout(resolve(_.bind(that.update_layout2), that), 0);
```

Comment:
I think this should be `setTimeout(function() {resolve(that.update_layout2())}, 0);`